File: anika.dovob/src/main/java/com/example/anika/homework/chapter7/NormalItem.java
package com.example.anika.homework.chapter7;

import jakarta.persistence.*;

@Entity(name = "Ch7NormalItem")
@Table(name = "normal_item_ch7")
@DiscriminatorValue("NORMAL")
public class NormalItem extends Item {

    @Column(nullable = false)
    private String sku;

    @Column(name = "stock_quantity", nullable = false)
    private int stockQuantity;

    protected NormalItem() {
    }

    public NormalItem(String name, long basePrice, String sku, int stockQuantity) {
        super(name, basePrice);
        this.sku = sku;
        this.stockQuantity = stockQuantity;
    }

    public String getSku() {
        return sku;
    }

    public int getStockQuantity() {
        return stockQuantity;
    }
}

File: gracie.lee/src/main/kotlin/homework/chapter_7/NormalItem.kt
package homework.chapter_7

import jakarta.persistence.*

@Entity(name = "Ch7NormalItemKt")
@Table(name = "normal_item_ch7_kt")
@DiscriminatorValue("NORMAL") // DTYPE에 들어갈 값
class NormalItem(
    @Column(nullable = false)
    var sku: String = "",

    @Column(nullable = false)
    var stockQuantity: Int = 0
) : Item()

// INSERT 시 2번 실행: item_ch7_kt에 공통 필드 + normal_item_ch7_kt에 고유 필드
// SELECT 시 JOIN 발생: item_ch7_kt + normal_item_ch7_kt